src/models/ontology_mapping_engine.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np
+from thefuzz import fuzz
+
+logger = src.CustomLogger.custom_logger.CustomLogger()
+
+
+
+class OntoMapEngine():
+    def __init__(self, 
+                 method:str, 
+                 query:list[str], 
+                 corpus:list[str],
+                 cura_map:dict, 
+                 topk:int=5, 
+                 yaml_path:str='method_model.yaml', 
+                 om_strategy='st') -> None:
+        """
+        This class is responsible for initializing the OntoMapEngine object.
+
+        Args:
+            method (str): The name of the method.
+            query (list): The list of queries.
+            corpus (list): The list of corpus.
+            topk (int): The number of top matches to return.
+            from_tokenizer (bool): A flag to indicate if the model is loaded from a tokenizer.
+            yaml_path (str): The path to the YAML file.
+        """
+        self.method = method
+        self.query = query
+        self.corpus = corpus
+        self.topk = topk
+        self.om_strategy = om_strategy
+        # self.from_tokenizer = from_tokenizer
+        self.yaml_path = yaml_path
+        self.cura_map = cura_map
+
+        self._logger = logger.custlogger(loglevel='INFO')
+        self._logger.info("Initialized OntoMap Engine module")
+
+
+    def _exact_matching(self):
+        """
+        This function is responsible for performing exact matching.
+
+        Here we want to separate exact matches of query to large corpus 
+        Returns:
+            list: The list of exact matches from query 
+        """
+        return [q for q in self.query if q in self.corpus] 
+    
+    def _fuzzy_matching(self, fuzz_ratio:int=80):
+        """
+        This function is responsible for performing fuzzy matching.
+
+        Here we want to separate fuzzy matches of query to large corpus 
+        Returns:
+           list(str): The list of exact matches from query  
+        """
+        return [q for q in self.query if fuzz.partial_ratio(q, self.corpus) > fuzz_ratio]
+
+    def _om_model_from_strategy(self, non_exact_query_list: list[str]):
+        """
+        This function is responsible for returning the OntoMap model based on the strategy
+        """
+        if self.om_strategy == 'lm':
+            return oml.OntoMapLM(method=self.method, query=non_exact_query_list, corpus=self.corpus, cura_map=self.cura_map, topk=self.topk, from_tokenizer=True, 
+                                 yaml_path=self.yaml_path)
+            
+        elif self.om_strategy == 'st':
+            return oms.OntoMapST(method=self.method, query=non_exact_query_list, corpus=self.corpus, cura_map=self.cura_map, topk=self.topk, from_tokenizer=False, 
+                                 yaml_path=self.yaml_path)
+        elif self.om_strategy == 'llm':
+            raise NotImplementedError("OntoMap LLM is not implemented yet")
+        else:
+            raise ValueError("om_strategy should be either 'st' or 'llm'")
```

Comment:
`lm` but not `llm`?

Reply:
Yes I have 3 modules for ontology mapping 1) sentence transformers 2) language models 3) LLM (which will be the RAG based framework)

Reply:
I meant a potential typo in line 80 - `om_strategy should be either 'st' or 'llm'`. Shouldn't it say `'st' or 'lm'`?